cyverse_allocation/tasks.py

```diff
@@ -107,22 +104,26 @@ def renew_allocation_sources(renewal_strategy=False, current_time=False):
             continue
         compute_allowed = args['compute_allowed']
         for allocation_source in AllocationSource.objects.filter(renewal_strategy=str(strategy)):
-            renew_allocation_source_for(compute_allowed, allocation_source, current_time)
-
+            renew_allocation_source_for(compute_allowed, allocation_source, current_time,
+                                        ignore_current_compute_allowed, dry_run)
 
-def renew_allocation_source_for(compute_allowed, allocation_source, current_time):
-    source_snapshot = AllocationSourceSnapshot.objects.filter(allocation_source=allocation_source)
-    if not source_snapshot:
-        raise Exception('Allocation Source %s cannot be renewed because no snapshot is available' % (
-            allocation_source.name))
-    source_snapshot = source_snapshot.last()
 
+def renew_allocation_source_for(compute_allowed, allocation_source, current_time, ignore_current_compute_allowed=False,
+                                dry_run=False):
     # carryover logic
     # remaining_compute = 0 if source_snapshot.compute_allowed - source_snapshot.compute_used < 0 else source_snapshot.compute_allowed - source_snapshot.compute_used
     # total_compute_allowed = float(remaining_compute + compute_allowed)
 
-    snapshot_compute_allowed = float(source_snapshot.compute_allowed)
-    total_compute_allowed = compute_allowed if snapshot_compute_allowed <= compute_allowed else snapshot_compute_allowed
+    if ignore_current_compute_allowed:
+        total_compute_allowed = compute_allowed
+    else:
+        source_snapshot = AllocationSourceSnapshot.objects.filter(allocation_source=allocation_source)
+        if not source_snapshot:
+            raise Exception('Allocation Source %s cannot be renewed because no snapshot is available' % (
+                allocation_source.name))
+        source_snapshot = source_snapshot.last()
```

Comment:
FWIW i'm pretty sure `source_snapshot = AllocationSourceSnapshot.objects.filter(allocation_source=allocation_source).last()` is an "always-safe" call.. will not result in a `DoesNotExist` error and would return either `AllocationSourceSnapshot` or `None`

Reply:
Good point.